test/oc_reporters_SUITE.erl

```diff
@@ -81,12 +91,78 @@ sequential_reporter(_Config) ->
     SortedNames = [ChildSpanName1, ChildSpanName1, SpanName1, SpanName1],
 
     Received = lists:map(fun(Name) ->
-                             receive
-                                 {span, #span{name = Name}} ->
-                                     Name
-                             after 5000 ->
-                                     undefined
-                             end
+                                 receive
+                                     {span, #span{name = Name}} ->
+                                         Name
+                                 after 5000 ->
+                                         undefined
+                                 end
                          end, SortedNames), %% receive order is undefined though
 
     ?assertMatch(SortedNames, lists:sort(Received)).
+
+zipkin_reporter(_Config) ->
+
+    Self = self(),
+
+    meck:new(httpc),
+    meck:expect(httpc, request,
+                fun (post, {"http://ct-host:9411/endpoint", [], "application/json", Content}, [], []) ->
+                        Self ! {ok, Content},
+                        {ok, {{ok, 202, ok}, ok, ok}};
+                    (_, _, _, _)  ->
+                        {ok, {{ok, 202, ok}, ok, ok}}
+                end),
+
+    try
+        Parent = oc_trace:start_span(<<"Parent">>, undefined),
+        Child = oc_trace:start_span(<<"span-name">>,
+                                    Parent,
+                                    #{attributes => #{<<"attr1">> => <<"val1">>,
+                                                      <<"attr_as_function">> =>
+                                                          fun () -> <<"val2">> end}}),
+        oc_trace:finish_span(Child),
+        oc_trace:finish_span(Parent),
+
+        ParentSpanId = iolist_to_binary(io_lib:format("~16.16.0b", [Parent#span_ctx.span_id])),
+        ParentTraceId = iolist_to_binary(io_lib:format("~32.16.0b", [Parent#span_ctx.trace_id])),
+
+        ChildSpanId = iolist_to_binary(io_lib:format("~16.16.0b", [Child#span_ctx.span_id])),
+
+        receive
+            {ok, Content} ->
+                [#{<<"annotations">> := [],
+                   <<"debug">> := false,
+                   <<"duration">> := _,
+                   <<"id">> := ChildSpanId,
```

Comment:
ps I would make some sort of literal test to ensure the id/parentId is lowerhex 16 length and traceId is lowerhex 32 length. sometimes folks have had small bugs like 31 character trace id